design/appfile-design.md

```diff
@@ -0,0 +1,337 @@
+# Appfile: Extensible, User-friendly Application Config Format
+
+- Owner: Hongchao Deng (@hongchaodeng)
+- Date: 10/14/2020
+- Status: Implemented
+
+## Table of Contents
+
+- [Intro](#intro)
+- [Goals](#goals)
+- [Proposal](#proposal)
+  - [Registration via Definition/Capability](#registration-via-definitioncapability)
+  - [Templating](#templating)
+  - [CLI/UI interoperability](#cliui-interoperability)
+  - [vela up](#vela-up)
+- [Examples](#examples)
+
+## Intro
+
+Vela supports a user-friendly `docker-compose` style config format called `Appfile`. It allows you to define an application's workloads and traits with an opinionated, simplified API interface.
+
+Here's an example to deploy a NodeJS express service:
```

Comment:
Could you also clarify a bit whether this appfile must be put under source code repo of the app or it's flexible (this is related to the `build` section).

Reply:
This is clarified in `vela up` section